srml/fees/src/lib.rs

```diff
@@ -14,7 +14,114 @@
+//! 
+//! ### Simple Code Snippet
+//! 
+//! Use the `ChargeFee` trait to charge and refund fees.
+//! 
+//! ```rust,ignore
+//! T::ChargeFee::charge_fee(transactor, fee);
+//! T::ChargeFee::refund_fee(transactor, fee);
+//! ```
+//! 
+//! ### Example from SRML
+//! 
+//! The `ChargeFee` trait is invoked in the `balances` and `contract` modules to manage fee-related logic in the following ways
+//! 1. `balances`: transfer/creation fee
+//! 2. `contract`: gas fee refund/buy
+//! 
+//! [In the balances module](https://github.com/paritytech/substrate/blob/687cf01ed34ee12d7aaf49bf99d7276192bc8363/srml/balances/src/lib.rs#L58), the `ChargeFee` trait is imported in the module's configuration trait:
+//! ```rust,ignore
+//! type ChargeFee: ChargeFee<Self::AccountId, Amount=Self::Balance>;
+//! ```
+//! 
+//! It is [later](https://github.com/paritytech/substrate/blob/687cf01ed34ee12d7aaf49bf99d7276192bc8363/srml/balances/src/lib.rs#L328) utilized to charge a transaction fee when the `transactor != dest` (any transfer betweeen different accounts):
+//! ```rust,ignore
+//! if transactor != dest {
+//!     T::ChargeFee::charge_fee(transactor, fee)?;
+//!     ...
+//! }
+//! ```
+//! 
+//! In the `contract` module, the `ChargeFee` trait is used to refund the gas fee ([ex](https://github.com/paritytech/substrate/pull/1815/files#diff-2179e1ee855613aa8f3343cf87154fe4R241)):
+//! ```rust,ignore
+//! let _ = <T as Trait>::ChargeFee::refund_fee(transactor, refund);
+//! ```
+//! 
+//! Moreover the following code in [substrate/node/runtime/src/lib.rs](https://github.com/paritytech/substrate/blob/687cf01ed34ee12d7aaf49bf99d7276192bc8363/node/runtime/src/lib.rs#L106) hooks the `fees` module and `balances` module, thereby enabling greater flexibility when implementing and extending the `ChargeFee` trait.
+//! ```rust,ignore
+//! impl balances::Trait for Runtime {
+//! 	type Balance = Balance;
+//! 	type OnFreeBalanceZero = ((Staking, Contract), Session);
+//! 	type OnNewAccount = Indices;
+//! 	type ChargeFee = fees::Module<Runtime>;
+//! 	type Event = Event;
+//! }
+//! ```
+//! 
+//! A [similar configuration](https://github.com/paritytech/substrate/blob/6ac1f183e0852a387953592d31f01957ff3c76f8/node/runtime/src/lib.rs#L111) involves utilizing the `balances` module in the `fees` implementation.
+//! ```rust,ignore
+//! impl fees::Trait for Runtime {
+//! 	type Event = Event;
+//! 	type TransferAsset = Balances;
+//! }
+//! ```
+//! If a chain does not have the `balances` module, then `TransferAsset` can be set to `()`
+//! ```rust,ignore
+//! type TransferAsset = ();
+//! ```
+//! In this case, no fee will be charged. Therefore, chains with different implementations of the balances module only need to implement the `TransferAsset` trait in order to be used by the `fees` module.
```

Comment:
well the therefore is not a direct implication no ?
I mean TransferAsset = () is used for testing afaik, and is not related to the fact that any module implementing TransferAsset can be used as associated type here